packages/xarc-app-dev/src/config/archetype.ts

```diff
@@ -15,8 +15,20 @@ let cachedArchetype = null;
  * @returns options - final options with defaults and env applied
  */
 module.exports = function getDevOptions(user: XarcOptions = {}) {
+  //checking for cwd from xclap or from env
+  const cwd = user.cwd || process.env.XARC_CWD || process.cwd();
+  process.env.XARC_CWD = cwd;
+
   if (cachedArchetype) {
+    // if cached is already runnig
     cachedArchetype._fromCache = true;
+    if (
```

Comment:
explain this

Reply:
I observed that this is needed because I did the changes to other files in xarc-app-dev and before our xclap.ts there is options.ts which makes a call and it save our xarcOptions.json. After that whenever this method is getting called from xclap.ts and if we pass our user ={cwd:"cwd  value"} then it checks for the cache and cache already exists then its not storing our value and return from there - This is the reason I check for cwd there and save it if its not saved